include/glow/Backends/Backend.h

```diff
@@ -44,10 +44,10 @@ class Backend {
   virtual ~Backend() = default;
 
   /// Prepare the interpreter for execution of new code.
-  virtual void init() = 0;
+  virtual void init(std::unique_ptr<const IRFunction> IR) = 0;
```

Comment:
Since the `unique_ptr` transfers ownership, there's no need for `const`. If `init` owns the IR, it can do whatever it wants to it, including mutating it.

Reply:
I kind of liked the `const` here as an indication that the backend/init doesn't modify the IR, even though ownership would indicate that it might.  So, for instance, if you `dump` it beforehand you know it's not being changed inside the backend.  But I'm not too attached to it, so I'll change it if you feel strongly.

Reply:
I think it would be reasonable for a backend to do its own transformations on the IR. That might delay the introduction of a backend-specific IR, or avoid it altogether.

Such a backend should provide command line options to dump the IR after the modifications.

Reply:
Sounds good.  Done!